scripts/ci/install_docker.sh

```diff
@@ -5,11 +5,7 @@ if [[ $(dpkg --print-architecture) == "amd64" ]]; then
     echo "Docker is already installed on AMD64"
     exit 0
 fi
-# https://docs.docker.com/engine/security/rootless/
-/bin/bash -c "$(curl -fsSL https://get.docker.com)"
-sudo apt-get install -y uidmap
-dockerd-rootless-setuptool.sh install
-export XDG_RUNTIME_DIR=/home/cloudtest/.docker/run
-PATH=/usr/bin:/sbin:/usr/sbin:$PATH dockerd-rootless.sh &
-sleep 5
-docker context use rootless
+
+# https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script
+curl -fsSL https://get.docker.com/ | sh
+sudo chmod 666 /var/run/docker.sock
```

Comment:
Use `chmod 666` to ensure `docker` can be used without `sudo` like runner-image agent.

In runner-image, this is achieved by adding user to `docker` group: https://github.com/actions/runner-images/blob/86ef9be8745af78c1e11cce2921532217216c90b/images/ubuntu/scripts/build/install-docker.sh#L49

But we can't re-login for it to take effect in the pipeline.

Reply:
`runner-image` is a general concept. I believe you are trying to mention [Microsoft-hosted agents](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent).

Reply:
I prefer we stick to the example at https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script, instead of piping to `| sh` in case something unexpected happens:

![image](https://github.com/user-attachments/assets/b27af79b-f2e9-4922-a2d3-c4ae43633f24)

Reply:
> Use `chmod 666` to ensure `docker` can be used without `sudo` like runner-image agent.

Any reference for this?